### PR TITLE
fix: Add missing private dns zones to applicable policy assignment and module 

### LINKED ISSUE #897: Azure Policy Assignment - Configure Azure PaaS services to use private DNS zones
### What happened? Provide a clear and concise description of the bug, including deployment details.

It looks like the Table Storage Private DNS Zone is missing from this policy. Is this still an issue as per the link below?

https://github.com/Azure/Enterprise-Scale/issues/1502

### Please provide the correlation id associated with your error or bug.

No error, missing DNS zone for table storage 

### What was the expected outcome?

Zone added

### Relevant log output

```Shell

```

### Check previous GitHub issues

- [x] I have searched the issues for this item and found no duplicate

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by oZakari**
# Overview/Summary

This pull request includes significant updates to the configuration and policy assignment files, primarily focusing on the management of private DNS zones and the addition of geographical codes.

### Configuration updates:

* [`.github/dependabot.yml`](diffhunk://#diff-dd4fbda47e51f1e35defb9275a9cd9c212ecde0b870cba89ddaaae65c5f3cd28R10-R12): Added a commit message prefix configuration for dependabot updates.
* [`accelerator/.config/ALZ-Powershell-Auto.config.json`](diffhunk://#diff-a5de3e24ccb80825a0d7b8004ca118e9856a96eecc107710e832a04f95b67880R307-R310): Added a new parameter for the location of private DNS zones.

### Policy assignment updates:

* [`infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep`](diffhunk://#diff-90b5177f0a982776fb0b5ac49607081d27a7285e686a0308d588520f1b89c0aeR101-R103): Added a new parameter for the location of private DNS zones and introduced a dictionary of geographical codes. [[1]](diffhunk://#diff-90b5177f0a982776fb0b5ac49607081d27a7285e686a0308d588520f1b89c0aeR101-R103) [[2]](diffhunk://#diff-90b5177f0a982776fb0b5ac49607081d27a7285e686a0308d588520f1b89c0aeR538-R667)
* [`infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep`](diffhunk://#diff-90b5177f0a982776fb0b5ac49607081d27a7285e686a0308d588520f1b89c0aeR538-R667): Expanded the list of private DNS zones to include additional Azure services.
* [`infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep`](diffhunk://#diff-90b5177f0a982776fb0b5ac49607081d27a7285e686a0308d588520f1b89c0aeL1965-L2123): Updated the policy assignment module to incorporate the newly added private DNS zones.
## Related Issues/Work Items

Fixes #897

## This PR fixes/adds/changes/removes

1. Adds all missing dns zones from policy set definition file

### Breaking Changes

None

## Testing Evidence

Replace this with any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [ ] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Review by jtracey93 — APPROVED** [review]

File: infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep [hunk 1]
```diff
@@ -532,60 +535,136 @@ var varPrivateDnsZonesResourceGroupSubscriptionId = !empty(parPrivateDnsResource
...
+  ukwest: 'ukw'
+  westcentralus: 'wcus'
+  westeurope: 'we'
+  westindia: 'inw'
+  westus: 'wus'
+  westus2: 'wus2'
+  westus3: 'wus3'
+}
+
+var varSelectedGeoCode = varGeoCodes[?parPrivateDnsZonesLocation] ?? 'changeme'
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep:598]: Change me?

File: infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep [hunk 2]
```diff
@@ -98,6 +98,9 @@ param parDdosProtectionPlanId string = ''
 @description('Resource ID of the Resource Group for Private DNS Zones. Empty to skip assigning the Deploy-Private-DNS-Zones policy.')
 param parPrivateDnsResourceGroupId string = ''
 
+@description('Location of Private DNS Zones.')
+param parPrivateDnsZonesLocation string = 'eastus'
```
Comment (jtracey93) [inline infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep:102]: Should this default to eastus? Or should it be a required input, for the same as the resource group or deployment location?

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep (hunk 2 above)
Comment (oZakari) [reply infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep:102]: Fixed in latest commit

File: infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep (hunk 1 above)
Comment (oZakari) [reply infra-as-code/bicep/modules/policy/assignments/alzDefaults/alzDefaultPolicyAssignments.bicep:598]: Fixed in latest commit

**Pull request merged**